codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/XmlShapeDeserVisitor.java

```diff
@@ -190,7 +190,7 @@ public void generateFlattenedCollectionDeserializer(GenerationContext context, C
                 getUnwrappedMapDelegateFunctionName(context, shape), symbol, () -> {
                     // initialize the output member variable
                     generatesIntializerForOutputVariable(context, shape);
-                    writer.openBlock("for {", "}", () -> {
+                    writer.openBlock(" switch { case true: ", "}", () -> {
```

Comment:
What's the reason for this being a `switch` rather then a `for`. Why have this in a block if its always going to be executed?

Reply:
We don't need a loop here, so replaced `for` with `switch`. 

The reason I opted for switch was to reuse the deser visitor methods. For simple shapes, we generate a break statement and break expects to be scoped within `for`, `switch` or `select` block.  

We can refactor this, if this affects performance (don't know how much it will affect, did this as a quick fix to unblock on customizations).

Reply:
Makes sense, I would personally think that generating it as:
```
switch {
default:
    // generated code
}
```
Would be clearer since there is value being compared against. So it looks less like an accidental generation error. At least in my opinion. Otherwise changes look good.

Reply:
Yeah I can change to using default. Looks much cleaner.

Reply:
updated